File: CfxGenerator/StructProperty.vb
'' Copyright (c) 2014-2015 Wolfgang Borgsmüller
'' All rights reserved.
'' 
'' Redistribution and use in source and binary forms, with or without 
'' modification, are permitted provided that the following conditions 
'' are met:
'' 
'' 1. Redistributions of source code must retain the above copyright 
''    notice, this list of conditions and the following disclaimer.
'' 
'' 2. Redistributions in binary form must reproduce the above copyright 
''    notice, this list of conditions and the following disclaimer in the 
''    documentation and/or other materials provided with the distribution.
'' 
'' 3. Neither the name of the copyright holder nor the names of its 
''    contributors may be used to endorse or promote products derived 
''    from this software without specific prior written permission.
'' 
'' THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS 
'' "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
'' LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
'' FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
'' COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
'' INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
'' BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS 
'' OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND 
'' ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR 
'' TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE 
'' USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.



Public Class StructProperty

    Public ReadOnly Getter As StructMember
    Public ReadOnly Setter As StructMember
    Public ReadOnly IsBoolean As Boolean
    Sub New(getter As StructMember, setter As StructMember, isBoolean As Boolean)
        Me.Getter = getter
        Me.Setter = setter
        Me.IsBoolean = isBoolean
        getter.IsProperty = True
        getter.Callback.PropertyName = PropertyName
        If setter IsNot Nothing Then
            setter.IsProperty = True
            setter.Callback.PropertyName = PropertyName
        End If
    End Sub

    ReadOnly Property PropertyName As String
        Get
            If Getter.Name.StartsWith("get_") Then
                Return Getter.PublicName.Substring(3)
            Else
                Return Getter.PublicName
            End If
        End Get
    End Property

End Class

File: CfxGenerator/StructProperty.cs
// Copyright (c) 2014-2017 Wolfgang Borgsmüller
// All rights reserved.
// 
// This software may be modified and distributed under the terms
// of the BSD license. See the License.txt file for details.

public class StructProperty {
    public readonly CefCallbackFunction Getter;
    public readonly CefCallbackFunction Setter;
    public readonly bool IsBoolean;

    public StructProperty(CefCallbackFunction getter, CefCallbackFunction setter, bool isBoolean) {
        this.Getter = getter;
        this.Setter = setter;
        this.IsBoolean = isBoolean;
        getter.IsProperty = true;
        getter.PublicPropertyName = PropertyName;
        if(setter != null) {
            setter.IsProperty = true;
            setter.PublicPropertyName = PropertyName;
        }
    }

    public string PropertyName {
        get {
            if(Getter.Name.StartsWith("get_")) {
                return Getter.PublicName.Substring(3);
            } else {
                return Getter.PublicName;
            }
        }
    }
}